src/aap_eda/core/validators.py

```diff
@@ -62,7 +62,7 @@ def check_if_de_exists(decision_environment_id: int) -> int:
     return decision_environment_id
 
 
-def check_if_de_valid(image_url: str, eda_credential_id: int):
+def check_if_de_valid(image_url: str, eda_credential_id: int = None):
```

Comment:
The type checking is not correct because `None` is not an int. You can use [typing.Optional](https://docs.python.org/3/library/typing.html#typing.Optional)
```suggestion
def check_if_de_valid(image_url: str, eda_credential_id: tp.Optional[int] = None):
```